settings/DevHome.Settings/ViewModels/PreferencesViewModel.cs

```diff
@@ -32,4 +71,19 @@ private async Task SwitchThemeAsync(ElementTheme theme)
             await _themeSelectorService.SetThemeAsync(theme);
         }
     }
+
+    partial void OnLaunchAtStartupChanged(bool value)
+    {
+        var startup = StartupTask.GetAsync(StartupTaskId).GetAwaiter().GetResult();
+        if (value && startup.State != StartupTaskState.Enabled)
+        {
+            LaunchAtStartupButtonEnabled = false;
+
+            var newState = startup.RequestEnableAsync().GetAwaiter().GetResult();
+            if (newState != StartupTaskState.Enabled)
+            {
+                LaunchAtStartup = false;
+            }
+        }
+    }

```

Comment:
make this async/await

Reply:
+1 for `async`
I believe this one is using the `partial` methods generated by he MVVM toolkit package.

Is there a way to make this `await`-able (CC: @Sergio0694), or it would work as expected if toggled quickly?

Two other ways come to mind to solve this (maybe there's yet another/better way?)
1. Implement an async handler for the [`Toggled` event](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.toggleswitch.toggled?view=windows-app-sdk-1.2) in `ToggleSwitch` and call an async method in the ViewModel with a bool param for on/off.
2. Use an `EventTriggerBehavior` and invoke a command for the `Toggled` event and pass the `IsOn` as param (this might be the old value, so inverse of the param is the new value)

Reply:
Yeah this is correct, the method is generated by the MVVM Toolkit, and it cannot return a `Task`, as it's called by the setter of the generated property, which cannot be asynchronous. I think 2 is the best solution here: make the `ToggleButton` just bind to `IsOn` in `OneWay` mode, and then have an `EventTriggerBehavior` for `Toggled` passing the current parameter, and a command to do this logic. You can then make that method with `[RelayCommand]` properly asynchronous, and it'll correctly map to an `IAsyncRelayCommand<T>` object, and things will work as expected (with the added bonus of eg. concurrency control, etc.) 🙂